static/src/javascripts/projects/commercial/modules/article-aside-adverts.spec.ts

```diff
@@ -45,38 +49,42 @@ describe('Standard Article Aside Adverts', () => {
 	});
 
 	it('should resolve immediately if the secondary column does not exist', (done) => {
-		if (document.body) {
-			document.body.innerHTML = `<div class="js-content-main-column"></div>`;
-		}
+		document.body.innerHTML = `<div class="js-content-main-column"></div>`;
 
-		init().then((resolve) => {
+		void init().then((resolve) => {
 			expect(resolve).toBe(false);
 			done();
 		});
 	});
 
 	it('should have the correct size mappings and classes', (done) => {
-		fastdomMeasureSpy.mockReturnValue(Promise.resolve([2000, 0]));
-		fakeMediator.once('page:defaultcommercial:right', (adSlot) => {
-			expect(adSlot.classList).toContain('js-sticky-mpu');
-			expect(adSlot.getAttribute('data-mobile')).toBe(
-				'1,1|2,2|300,250|300,274|300,600|fluid',
-			);
-			done();
-		});
-		init();
+		mockMeasure(2000, 0);
+		fakeMediator.once(
+			'page:defaultcommercial:right',
+			(adSlot: HTMLElement) => {
+				expect(adSlot.classList).toContain('js-sticky-mpu');
+				expect(adSlot.getAttribute('data-mobile')).toBe(
+					'1,1|2,2|300,250|300,274|300,600|fluid',
+				);
+				done();
```

Comment:
Not your code but Is it possible to change the uses of `done()` to `async () => {` and then `await` for any promise calls.

To my eyes it reads a bit easier.

Examples here: https://jestjs.io/docs/tutorial-async

Reply:
async / await seem more straightforward to me too but I'm struggling to figure out how to use them here.
How can we define `adSlot` if not as a parameter of `once`'s listener function?

Reply:
Assuming that `init()` has its promises properly chained I _think_ you should be able to:

1. remove the `done()` function,
1. `await init()`
1. make the test function `async` so Jest knows to await the function

This test passes an actual function which internally wraps test assertions.

However usually for callbacks you'd want to pass a mock function i.e. `jest.fn()` in place of the callback, then you can assert later on what arguments it was called with.

Reply:
Thanks, that makes sense - hows that?